readthedocs/doc_builder/director.py

```diff
@@ -197,17 +206,36 @@ def build(self):
 
     # VCS checkout
     def checkout(self):
-        log.info(
-            "Clonning repository.",
-        )
+        """Checkout Git repo and load build config file."""
+
+        log.info("Cloning repository.")
         self.vcs_repository.update()
 
         identifier = self.data.build_commit or self.data.version.identifier
         log.info("Checking out.", identifier=identifier)
         self.vcs_repository.checkout(identifier)
 
-        self.data.config = load_yaml_config(version=self.data.version)
+        # The director is responsible for understanding which config file to use for a build.
+        # In order to reproduce a build 1:1, we may use readthedocs_yaml_path defined by the build
+        # instead of per-version or per-project.
+        # Use the below line to fetch the readthedocs_yaml_path defined per-build.
+        # custom_config_file = self.data.build.get("readthedocs_yaml_path", None)
+        custom_config_file = None
+
+        # This logic can be extended with version-specific config files
+        if not custom_config_file and self.data.version.project.readthedocs_yaml_path:
+            custom_config_file = self.data.version.project.readthedocs_yaml_path
+
+        if custom_config_file:
+            log.info(
+                "Using a custom config file.", custom_config_file=custom_config_file
+            )
+        self.data.config = load_yaml_config(
+            version=self.data.version,
+            readthedocs_yaml_path=custom_config_file,
+        )
         self.data.build["config"] = self.data.config.as_dict()
+        self.data.build["readthedocs_yaml_path"] = custom_config_file
```

Comment:
This could be a method of this class, `load_config()` or similar. Not a blocker, tho.

Reply:
I'll leave that for the person that might find `checkout` too long... which might be _me_ in the future, in case we decide to add support for per-version or per-build `.readthedocs.yaml` paths :)